src/middleware/validator.js

```diff
@@ -1,95 +1,31 @@
 const logger = require('../logger')
+const config = require('../config')
 
-const createValidationSchema = validationMap => {
-  if (!validationMap) {
-    throw new Error('No validation rules supplied')
-  }
-
-  const validations = validationMap
-  const schemaObject = {}
-
-  if (validations) {
-    Object.keys(validations).forEach(key => {
-      let rule = validations[`${key}`]
-      switch (rule.type) {
-        case 'string':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.string().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.string().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.string()
-          }
-          break
-
-        case 'number':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.number().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.number().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.number()
-          }
-          break
-
-        case 'boolean':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.boolean().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.boolean().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.boolean()
-          }
-          break
-
-        case 'array':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.array().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.array().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.array()
-          }
-          break
+const configurations = config.getConfig()
 
-        case 'object':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.object().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.object().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.object()
-          }
-          break
-
-        default:
-          throw new Error(`Validation rule type is not supported: ${rule.type}`)
-      }
-    })
-
-    if (schemaObject) {
-      return Joi.object(schemaObject)
-    }
-
-    throw new Error('Joi validation schema creation failed')
+const performValidation = (ctx, schema) => {
+  if (!schema) {
+    throw new Error(`No validation rules supplied`)
   }
 
-  throw new Error('No validation schema in file')
-}
+  if (!ctx || !ctx.request || !ctx.request.body) {
+    throw new Error(`Invalid request body`)
+  }
 
-const performValidation = (ctx, schema) => {
-  const {error, value} = schema.validate(ctx.request.body)
+  const ajv = new Ajv({
+    nullable: configurations.nullable
+  })
+  const valid = ajv.validate(schema, ctx.request.body)
 
-  if (error) {
-    throw new Error(`Validation execution failed: ${error.message}`)
+  if (!valid) {
```

Comment:
My feeling is that if something errors out we obv need to log it as an error, but when something passes a middleware method it also needs to log an info message to inform us it has progressed to the next middleware. This might seem verbose but its useful to track the entire request has successfully processed. Otherwise we might say that we dont log anything when its successfull and we might have a log that never shows any interactions as they were all successful

Reply:
:+1: Awesome.

Reply:
fixed in this https://github.com/jembi/openhim-mediator-mapping/pull/21/commits/0f8e33385fb1476bbfce85be9e94863c63b4caae

Reply:
@bradsawadye This log message is duplicated as the log for this already existed here: https://github.com/jembi/openhim-mediator-mapping/pull/21/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json#diff-e08f7fe18f9f337f2dcdf8bec8a69a7bR33

My question here was more aimed at will this be spamming the logs in general and possibly to add the OpenHIM transaction ID to our log messages to be able to track the request where ever it is.

Reply:
reverted the commit